File: crawler/src/model/track.js
class Track {
    /**
     * @param {string?} musicBrainzId - Track id from musicBrainz api
     * @param {string} name - The title name from the telegram channel
     * @param {string?} musicBrainzName - The title name from the musicBrainzApi
     * @param {string} messageId - The messageId from telegram channel
     */
    constructor(name, musicBrainzId, musicBrainzName, messageId) {
        this.name = name;
        this.musicBrainzId = musicBrainzId;
        this.musicBrainzName = musicBrainzName;
        this.messageId = messageId;
    }

}

module.exports = Track;

File: api/src/main/java/com/citytube/api/models/Track.java
package com.citytube.api.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;


@EqualsAndHashCode(callSuper = true)
@Data
@SuperBuilder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "track")
public class Track extends BaseEntity{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String musicBrainzId;

    private String musicBrainzName;

    @Column(nullable = false)
    private Integer messageId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "album_id", nullable = false)
    private Album album;

}
